sdcm/nemesis.py

```diff
@@ -463,6 +463,21 @@ def disrupt_rolling_restart_cluster(self):
         self._set_current_disruption('RollingRestartCluster %s' % self.target_node)
         if not self._is_it_on_kubernetes():
             raise UnsupportedNemesis('RollingRestartCluster is supported only for kubernetes backends')
+
+        # Switch to com.scylladb.auth.SaslauthdAuthenticator
+        if self.cluster.params.get('prepare_saslauthd'):
+            if 'PasswordAuthenticator' in self.cluster.params.get('authenticator'):
+                for node in self.cluster.nodes:
+                    node.remoter.run(
+                        "sed -ie 's/^authenticator:.*/authenticator: com.scylladb.auth.SaslauthdAuthenticator/g' /etc/scylla/scylla.yaml")
+        self.cluster.rollout_restart()
+
+        # Switch back to PasswordAuthenticator
+        if self.cluster.params.get('prepare_saslauthd'):
+            if 'PasswordAuthenticator' in self.cluster.params.get('authenticator'):
+                for node in self.cluster.nodes:
+                    node.remoter.run(
+                        "sed -ie 's/^authenticator:.*/authenticator: PasswordAuthenticator/g' /etc/scylla/scylla.yaml")
```

Comment:
Why is it part of disrupt_rolling_restart_cluster?
Add it as a new nemesis.

Reply:
> Why is it part of disrupt_rolling_restart_cluster?
> Add it as a new nemesis.

I didn't want to repeat the rolling_restart nemesis again.
If prepare_saslauthd isn't enabled, the new nemesis will be completely same as rolling_restart nemesis.
However, I will add a new nemesis.

Reply:
Update: I added a new nemesis, and only restart scylla-server after authenticator switch.